duckduckgo/macos-browser
# Fixes startup options reset

Task/Issue URL: https://app.asana.com/0/0/1205465612515839/f

BSK PR: https://github.com/duckduckgo/BrowserServicesKit/pull/489
iOS PR: https://github.com/duckduckgo/iOS/pull/2003

## Description

Fixes startup options reset, and improves the options loading logic so we can easily unit test it.

## Testing

### Test the connection tester

1. Run the app
2. Try disabling the connection using the script below and `sudo pfctl -ef ~/Desktop/ruleset.txt`

[ruleset.txt](https://github.com/duckduckgo/macos-browser/files/12576052/ruleset.txt)

3. Ensure the tester properly flags the connection as being broken.

### Test the connection

1. Run the app
2. Connect NetP and make sure the connection works fine.
3. Disconnect
4. Delete all auth tokens.
5. Enable NetP again and make sure the connection works fine.

---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
